src/pages/distroSettings/tabs/ProviderTab/transformers.ts

```diff
@@ -1,22 +1,37 @@
 import { DistroSettingsTabRoutes } from "constants/routes";
 import { Provider } from "gql/generated/types";
 import { FormToGqlFunction, GqlToFormFunction } from "../types";
-import { staticProviderSettings } from "./transformerUtils";
+import {
+  staticProviderSettings,
+  dockerProviderSettings,
+} from "./transformerUtils";
 
 type Tab = DistroSettingsTabRoutes.Provider;
 
 export const gqlToForm = ((data) => {
   if (!data) return null;
 
-  const { provider, providerSettingsList } = data;
+  const { containerPool, provider, providerSettingsList } = data;
 
   switch (provider) {
     case Provider.Static:
       return {
         provider: {
           providerName: Provider.Static,
         },
-        ...staticProviderSettings(providerSettingsList[0]).form,
+        providerSettings: {
+          ...staticProviderSettings(providerSettingsList[0]).form,
+        },
+      };
+    case Provider.Docker:
+      return {
+        provider: {
+          providerName: Provider.Docker,
+        },
+        providerSettings: {
+          ...dockerProviderSettings(providerSettingsList[0]).form,
```

Comment:
I switched the `localhost` distro to use Docker and then tried to switch it back to static and saw this error:
<img width="412" alt="image" src="https://github.com/evergreen-ci/spruce/assets/9298431/73ad1bd1-f995-487d-b3ab-2a6eb601d88a">

It seems like there's validation [here](https://github.com/evergreen-ci/evergreen/blob/00e0aa6f50e98304d251b3afa1108840b6538641/model/distro/distro.go#L550-L552) preventing container pool distros from using a different provider.

Maybe someone on app would have insight into the best approach here. Do we want to ban Docker distros from having their types changed? Or do we maybe want to update the save function to remove this distro from the list of Docker distros? 🤔

Reply:
oh that's probably my fault. Have to clear the `containerPool` field in the transformers